clash-cores/src/Clash/Cores/LineCoding8b10b.hs

```diff
@@ -0,0 +1,103 @@
+{-# LANGUAGE CPP #-}
+
+-- |
+--   Copyright   :  (C) 2024, QBayLogic B.V.
+--   License     :  BSD2 (see the file LICENSE)
+--   Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+--
+--   8b/10b encoding and decoding functions
+module Clash.Cores.LineCoding8b10b where
+
+import qualified Clash.Cores.LineCoding8b10b.Decoder as Dec
+import qualified Clash.Cores.LineCoding8b10b.Encoder as Enc
+import Clash.Prelude
+
+-- | Data type that contains a 'BitVector' with the corresponding error
+--   condition of the decode function
+data Symbol8b10b
+  = -- | Correct data word
+    Dw (BitVector 8)
+  | -- | Correct control word
+    Cw (BitVector 8)
+  | -- | Incorrect data word
+    DwError (BitVector 8)
+  | -- | Running disparity error
+    RdError (BitVector 8)
+  deriving (Generic, NFDataX, Eq, Show)
+
+-- | Function to check whether a 'Symbol8b10b' results in a data word
+isDw :: Symbol8b10b -> Bool
+isDw (Dw _) = True
+isDw _ = False
+
+-- | Function to check whether a 'Symbol8b10b' is not an error value
+isValidSymbol :: Symbol8b10b -> Bool
+isValidSymbol (Cw _) = True
+isValidSymbol dw = isDw dw
+
+-- | Function to convert a 'Symbol8b10b' to a plain 'BitVector 8'
+fromDw :: Symbol8b10b -> BitVector 8
+fromDw dw = case dw of
+  Dw _dw -> _dw
+  Cw _dw -> _dw
+  _ -> 0
```

Comment:
That's reasonable, I took your suggestion. The thing is that you don't really _want_ to do anything with a value that has an `DwError` or `RdError`, so that's why I set it to be 0, but I guess it doesn't hurt if I don't.